extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -2015,6 +2016,11 @@ private boolean withinMinMaxRecordTime(final InputRow row)
     final boolean afterMaximumMessageTime = ioConfig.getMaximumMessageTime().isPresent()
                                             && ioConfig.getMaximumMessageTime().get().isBefore(row.getTimestamp());
 
+    if (!Intervals.ETERNITY.contains(row.getTimestamp())) {
+      log.error("Encountered row with timestamp that cannot be represented as a long: [%s]", row);
+      return false;
+    }
```

Comment:
how about reporting it as ParseException in case user has set reportParseExceptions to true ?

Reply:
changed this to throw a ParseException if reportParseExceptions is true